toolchains/make-reproducible-bootstrap.patch

```diff
@@ -0,0 +1,16 @@
+diff --git Makefile.in Makefile.in
+index 9535058..0b12e99 100644
+--- Makefile.in
++++ Makefile.in
+@@ -1042,8 +1042,8 @@ make_LDADD = $(LIBOBJS) $(GUILE_LIBS) lib/libgnu.a $(GETLOADAVG_LIBS) \
+ 		@LIBINTL@
+
+ AM_CPPFLAGS = -Isrc -I$(top_srcdir)/src -Ilib -I$(top_srcdir)/lib \
+-	-DLIBDIR=\"$(libdir)\" -DINCLUDEDIR=\"$(includedir)\" \
+-	-DLOCALEDIR=\"$(localedir)\" $(am__append_2)
++	-DLIBDIR=\".\" -DINCLUDEDIR=\"external/rules_foreign_cc/toolchains/make/include\" \
++	-DLOCALEDIR=\"external/rules_foreign_cc/toolchains/make/share/locale\" $(am__append_2)
```

Comment:
I appreciate this actually makes this reproducible but does this actually work to allow make to correctly find its locale directory? The configure_make rules run in `$BUILDTMP` which is below the bazel execroot and so this relative path is not going to be correct from that directory. 
I don't have a good solution for this though that is any better than what you've already done here; I suspect we don't really care that the make tool can't find its translations but we should at least try to ensure that the tool does work (Which I appreciate even in the current configuration isn't the case due to sandbox paths, but it would currently at least currently work in a non-sandboxed local execution environment).

Reply:
From what I can tell, these directories default to `/lib`, `/include` and `/share/locale`, so they are not `make`-specific. Based on a cursory look into the code, these variables are used for the following purposes:
* `LIBDIR` is only used in `remake.c`, apparently to find out whether a certain system library has changed since the last `make` invocation. This should not be needed in the context of `rules_foreign_cc`.
* `INCLUDEDIR` is the default search path for `Makefile`s to include, which can also be specified with the `-I` option to `make`. Picking up `Makefile`s from the host in unsandboxed builds sounds really risky, so maybe this should always be set to something like `/does/not/exist`.
* `LOCALEDIR` is used to make `make` translatable.

While I do favor including this patch as it helps reproducibility and these directories are already non-existent in current sandboxed builds, I can drop it if you prefer.